### PR TITLE
[generator2] provide a simpler alternative to StandaloneSetup

### PR DESCRIPTION
**Pull request by oehme**

### CODE REVIEW

File: plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorStandaloneSetup.xtend [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+			delegate.addProjectMapping(new ProjectMapping => [
+				projectName = mapping.key
+				path = mapping.value.path
+			])
+		]
+	}
+
+	private def getProjectMappings() {
+		#[
+			projectConfig.runtimeProjectName -> projectConfig.runtimeRoot,
```
Comment (szarnekow) [inline plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorStandaloneSetup.xtend]: Maybe I don't get it, but how would a client extend this, e.g. to register more projects - let's say for netbeans integration?

Comment (oehme) [inline plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorStandaloneSetup.xtend]: I'll make those methods protected.

Comment (oehme) [inline plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorStandaloneSetup.xtend]: Also regarding 'more projects', we are thinking about giving more structure to XtextProjectConfig to make that easier. So then you could say something like project.allSubProjects.map[root]...

**Comment by szarnekow** [comment]
StandaloneSetup#registerGeneratedEPackage was a nice and easy way to register a bunch of interdependent packages (they would be initialized transitively). How would I populate the registry with the simple API? Still fallback to the mwe StandaloneSetup I assume?

**Comment by oehme** [comment]
Yes, the old StandaloneSetup will be around and we can add that method to the new one if there is demand. But I'd like to keep it simple for now.

**Pull request merged**